Добавили 14 тестов по тестированию основной страницы и страницы заказ…

KasimovN opened the pull request:
…а приложения самока.

Тест вопросов/ответов на главной странице
Тест перехода на страницу заказ через верхнюю и нижнюю кнопку Заказать главной страницы. Тест создания заказа с 2-ми наборами данных
Тест перехода кликом на логотип Самокат
Тест перехода кликом на логотип Yandex

KasimovN committed:
Добавили 14 тестов по тестированию основной страницы и страницы заказа приложения самока.
Тест вопросов/ответов на главной странице
Тест перехода на страницу заказ через верхнюю и нижнюю кнопку Заказать главной страницы.
Тест создания заказа с 2-ми наборами данных
Тест перехода кликом на логотип Самокат
Тест перехода кликом на логотип Yandex

klushcheva commented on locators/main_page_locators.py:
```diff
@@ -0,0 +1,12 @@
+
+class MainPageLocators:
+
+    UPPER_ORDER_BUTTON = (By.XPATH, '//*[@class ="Button_Button__ra12g" and text() ="Заказать"]')
```
Можно лучше: потенциально нестабильный локатор. Вот эта цифро-буквенная часть класса в конце строки Button_Button__**ra12g** сгенерирована автоматически при деплое. При последующих апдейтах локатор может сломаться. Нужно использовать contains и только основное имя класса

KasimovN replied:
Исправил, локатор сделал через стабильного родителя.

klushcheva commented on pages/main_page.py:
```diff
@@ -0,0 +1,26 @@
+from locators.main_page_locators import MainPageLocators
+
+
+class MainPage:
```
Нужно поправить: в Page Object подходе мы сначала описываем BasePage, там все основные методы. И от BasePage наследуем остальные необходимые классы

KasimovN replied:
Исправил, для данной задачи думал не имеет смысла создавать отдельную сущность BasePage.

klushcheva commented on tests/test_order_page.py:
```diff
@@ -0,0 +1,68 @@
+        upper_order_button = OrderPage(driver)
+        upper_order_button.click_cookie_button()
+        upper_order_button.wait_and_find_element(OrderPageLocators.UPPER_ORDER_BUTTON).click()
+        assert driver.current_url == 'https://qa-scooter.praktikum-services.ru/order'
```
Можно лучше: стараемся избегать любого хардкода, выносим тексты в отдельный файл

KasimovN replied:
Исправлено.

klushcheva commented on tests/test_order_page.py:
```diff
@@ -0,0 +1,68 @@
+        yandex_logo.click_element(OrderPageLocators.YANDEX_LOGO)
+        windows = driver.window_handles
+        driver.switch_to.window(windows[1])
+        yandex_logo.wait_and_find_element((By.XPATH, '//*[text()="Найти"]'))
```
Нужно поправить: не хардкодим локаторы

KasimovN replied:
Исправил

klushcheva commented on tests/test_questions_answers.py:
```diff
@@ -0,0 +1,21 @@
+class TestQuestionsAnswers:
+    @allure.title('Проверка вопроса/ответа')
+    @allure.description('Проверка на тексты вопросов/ответов и на сопоставления ответов нужным вопросам')
+    @pytest.mark.parametrize('question, answer', Helper.parametrize_massive_qa())
```
Отлично: верно использована параметризация

KasimovN replied:
Спасибо.

KasimovN committed:
Добавили 14 тестов по тестированию основной страницы и страницы заказа приложения самока.
Тест вопросов/ответов на главной странице
Тест перехода на страницу заказ через верхнюю и нижнюю кнопку Заказать главной страницы.
Тест создания заказа с 2-ми наборами данных
Тест перехода кликом на логотип Самокат
Тест перехода кликом на логотип Yandex.
